mmedit/models/video_interpolater/basic_interpolater.py

```diff
@@ -0,0 +1,308 @@
+            if isinstance(iteration, numbers.Number):
+                save_path = osp.join(save_path,
+                                     f'{img_name}-{iteration + 1:06d}.png')
+            elif iteration is None:
+                save_path = osp.join(save_path, f'{img_name}.png')
+            else:
+                raise ValueError('iteration should be number or None, '
+                                 f'but got {type(iteration)}')
+            mmcv.imwrite(tensor2img(output), save_path)
+        elif output.ndim == 5:  # a sequence
+            folder_name = meta[0]['key'].split('/')[0]
+            for i in range(0, output.size(1)):
+                if isinstance(iteration, numbers.Number):
+                    save_path_i = osp.join(save_path, folder_name,
+                                           f'{i:08d}-{iteration + 1:06d}.png')
+                elif iteration is None:
+                    save_path_i = osp.join(save_path, folder_name,
+                                           f'{i:08d}.png')
+                else:
+                    raise ValueError('iteration should be number or None, '
+                                     f'but got {type(iteration)}')
+                mmcv.imwrite(tensor2img(output[:, i, :, :, :]), save_path_i)
+
+    def forward_dummy(self, img):
+        """Used for computing network FLOPs.
+
+        Args:
+            img (Tensor): Input frames.
+
+        Returns:
+            Tensor: Output frame(s).
+        """
+        out = self.generator(img)
+        return out
+
+    def train_step(self, data_batch, optimizer):
+        """Train step.
+
+        Args:
+            data_batch (dict): A batch of data.
+            optimizer (obj): Optimizer.
+
+        Returns:
+            dict: Returned output.
+        """
+        outputs = self(**data_batch, test_mode=False)
+        loss, log_vars = self.parse_losses(outputs.pop('losses'))
+
+        # optimize
+        optimizer['generator'].zero_grad()
+        loss.backward()
+        optimizer['generator'].step()
+
+        outputs.update({'log_vars': log_vars})
+        return outputs
+
+    def val_step(self, data_batch, **kwargs):
+        """Validation step.
+
+        Args:
+            data_batch (dict): A batch of data.
+            kwargs (dict): Other arguments for ``val_step``.
+
+        Returns:
+            dict: Returned output.
+        """
+        output = self.forward_test(**data_batch, **kwargs)
+        return output
+
+    @staticmethod
+    def split_frames(input_tensors):
+        """split input tensors for inference.
+
+        This is a basic function, interpolate a frame between the given two
+        frames.
+
+        Args:
+            input_tensors (Tensor): Tensor of input frames.
+
+        Returns:
+            list[Tensor]: Splided tensors.
+        """
+
+        num_frames = input_tensors.shape[0]
+
+        result = []
+        for i in range(0, num_frames - 1, 1):
+            result.append(input_tensors[i:i + 2])
```

Comment:
Suggestions:
1. We can follow your previous approach of using `num_input_frames` as one argument. In this way, CAIN (2 frames->1 frame) and QVI (4 frames->1 frame) can both use this class. The argument `step` can be omitted.
2. We don't need to make it a static method. In such a way, `self.num_input_frames` can be passed to this function, if the first point is considered.